contracts/TreasuryExtender.sol

```diff
@@ -0,0 +1,282 @@
+pragma solidity ^0.8.10;
+
+// interfaces
+import "./interfaces/IERC20.sol";
+import "./interfaces/ITreasury.sol";
+import "./interfaces/IAllocator.sol";
+import "./interfaces/ITreasuryExtender.sol";
+
+// types
+import "./types/OlympusAccessControlledImproved.sol";
+
+// libraries
+import "./libraries/SafeERC20.sol";
+
+contract TreasuryExtender is OlympusAccessControlledImproved, ITreasuryExtender {
+    using SafeERC20 for IERC20;
+
+    ITreasury immutable treasury;
+
+    IAllocator[] public allocators;
+
+    mapping(IAllocator => AllocatorData) public allocatorData;
+
+    uint256 private totalValueAllocated;
+
+    constructor(address treasuryAddress, address authorityAddress)
+        OlympusAccessControlledImproved(IOlympusAuthority(authorityAddress))
+    {
+        treasury = ITreasury(treasuryAddress);
+        allocators.push(IAllocator(address(0)));
+    }
+
+    //// "MODIFIERS"
+
+    function _allocatorActivated(AllocatorStatus status) internal pure {
+        require(AllocatorStatus.ACTIVATED == status, "TreasuryExtender::AllocatorOffline");
+    }
+
+    function _allocatorOffline(AllocatorStatus status) internal pure {
+        require(AllocatorStatus.OFFLINE == status, "TreasuryExtender::AllocatorActivated");
+    }
+
+    function _onlyAllocator(uint256 id, address sender) internal view {
+        require(IAllocator(sender) == allocators[id], "TreasuryExtender::OnlyAllocator");
+    }
+
+    //// FUNCTIONS
+
+    function registerAllocator(address newAllocatorAddress) external override {
```

Comment:
I've done this for the reason to make the external functions standardized... it will work the same, just this way, it's explicitly clear for _any_ reader that the input is an address.